tools/admin/src/main/scala/whisk/core/cli/Main.scala

```diff
@@ -0,0 +1,163 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.cli
+
+import java.io.File
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import ch.qos.logback.classic.{Level, LoggerContext}
+import org.rogach.scallop._
+import org.slf4j.LoggerFactory
+import pureconfig.error.ConfigReaderException
+import whisk.common.{AkkaLogging, Logging, TransactionId}
+import whisk.core.database.{LimitsCommand, UserCommand}
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+import scala.util.{Failure, Success, Try}
+
+class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
+  banner("OpenWhisk admin command line tool")
+
+  val verbose = tally()
+  val configFile = opt[File](descr = "application.conf path")
+  printedName = Main.printedName
+
+  addSubcommand(new UserCommand)
+  addSubcommand(new LimitsCommand)
+  shortSubcommandsHelp()
+
+  requireSubcommand()
+  validateFileExists(configFile)
+  verify()
+}
+
+object Main {
+  val printedName = "wskadmin"
+
+  def main(args: Array[String]) {
+    //Parse conf before instantiating actorSystem to ensure fast pre check of config
+    val conf = new Conf(args)
+    initLogging(conf)
+    initConfig(conf)
+
+    conf.subcommands match {
+      case List(c: WhiskCommand) => c.failNoSubCommand()
+      case _                     =>
+    }
+    val exitCode = execute(conf)
+    System.exit(exitCode)
+  }
+
+  private def execute(conf: Conf): Int = {
+    implicit val actorSystem = ActorSystem("admin-cli")
+    try {
+      executeWithSystem(conf)
+    } finally {
+      actorSystem.terminate()
+      Await.result(actorSystem.whenTerminated, 30.seconds)
+    }
+  }
+
+  private def executeWithSystem(conf: Conf)(implicit actorSystem: ActorSystem): Int = {
+    implicit val logger = new AkkaLogging(akka.event.Logging.getLogger(actorSystem, this))
+    implicit val materializer = ActorMaterializer.create(actorSystem)
+
+    val admin = new WhiskAdmin(conf)
+    val result = Try {
+      val f = admin.executeCommand()
+      Await.result(f, 30.seconds)
+    }
+    result match {
+      case Success(r) =>
+        r match {
+          case Right(msg) =>
+            println(msg)
+            0
+          case Left(e) =>
+            printErr(e.message)
+            e.code
+        }
+      case Failure(e) =>
+        e match {
+          case _: ConfigReaderException[_] =>
+            printErr("Incomplete config. Provide application.conf via '-c' option")
+          case _ =>
+        }
+        e.printStackTrace()
+        3
```

Comment:
2 for syntax/usage error?

Reply:
Yes. It tries to map to existing `wskadmin` error 

```scala
class CommandError(val message: String, val code: Int)
case class IllegalState(override val message: String) extends CommandError(message, 1)
case class IllegalArg(override val message: String) extends CommandError(message, 2)
```